feat: MPSNGM-294 patch DA network variables

EmandM opened the pull request:
<!-- Replace this block with what this PR does and why. Describe what you'd like reviewers to know, how you applied the engineering principles, and any interesting tradeoffs made. Delete bullet points below that don't apply, and update the changelog section as appropriate. -->

<!-- Add short version of the JIRA ticket to the PR title (e.g. "feat: new shiny feature [MTT-123]") -->
[MPSNGM-294](https://jira.unity3d.com/browse/MPSNGM-294)

<!-- Add RFC link here if applicable. -->

## Changelog

- Fixed: re-added support for various network variable types within DA


<!--  Uncomment and mark items off with a * if this PR deprecates any API:
### Deprecated API
- [ ] An `[Obsolete]` attribute was added along with a `(RemovedAfter yyyy-mm-dd)` entry.
- [ ] An [api updater] was added.
- [ ] Deprecation of the API is explained in the CHANGELOG.
- [ ] The users can understand why this API was removed and what they should use instead.
-->

EmandM committed:
feat: MPSNGM-294 patch DA network variables

NoelStephensUnity commented on com.unity.netcode.gameobjects/Runtime/NetworkVariable/Collections/NetworkList.cs:
```diff
@@ -1,5 +1,6 @@
 using System;
 using System.Collections.Generic;
+using System.Runtime.Serialization;
```
Is this namespace 100% supported in Unity/Mono?

NoelStephensUnity committed:
Merge branch 'develop-2.0.0' into feat/mpsngm-294-fix-da-network-variables

NoelStephensUnity committed:
fix

Fixing merge issue.
Updating some methods where it was using writer as opposed to reader.

NoelStephensUnity commented on testproject/ProjectSettings/MultiplayerManager.asset:
```diff
@@ -0,0 +1,7 @@
+%YAML 1.1
```
Don't add this asset or the Settings.json.

NoelStephensUnity committed:
style

adding whitespace.

NoelStephensUnity committed:
style

removing System.Runtime.Serialization using directive as it is not needed.

NoelStephensUnity commented:
So far it is looking good.
Let me know when you finish and I can do a second pass.

EmandM committed:
Merge branch 'develop-2.0.0' of https://github.com/Unity-Technologies/com.unity.netcode.gameobjects into feat/mpsngm-294-fix-da-network-variables

EmandM committed:
Remove unnecessary files

NoelStephensUnity committed:
Merge branch 'develop-2.0.0' into feat/mpsngm-294-fix-da-network-variables

NoelStephensUnity commented on testproject/Packages/packages-lock.json:
```diff
@@ -1,21 +1,22 @@
   "dependencies": {
     "com.unity.addressables": {
-      "version": "1.21.9",
+      "version": "2.0.8",
```
You an exclude this from the PR.

NoelStephensUnity reviewed (approved):
Looks good Emma!
